pkg/cmd/codespace/ssh.go

```diff
@@ -276,13 +282,21 @@ func selectSSHKeys(
 		return autoKeyPair, true, nil
 	}
 
-	keyPair, err := firstConfiguredKeyPair(ctx, customConfigPath, opts.profile)
-	if err != nil {
-		if !errors.Is(err, errKeyFileNotFound) {
-			return nil, false, fmt.Errorf("checking configured keys: %w", err)
+	var keyPair *ssh.KeyPair
+	var err error
+
+	if !generateKeys {
+		keyPair, err = firstConfiguredKeyPair(ctx, customConfigPath, opts.profile)
+		if err != nil {
+			if !errors.Is(err, errKeyFileNotFound) {
+				return nil, false, fmt.Errorf("checking configured keys: %w", err)
+			}
+			generateKeys = true
 		}
+	}
 
-		// no valid key in ssh config, generate one
+	// user wants to force generate keys or no valid key in ssh config, generate one
+	if generateKeys {
```

Comment:
Good suggestion. I'd bet there isn't a case where the return for `firstConfiguredKeyPair` returns `nil, nil`, but adding the check is a good defensive check against that case if it happens in the future (especially since that would be something a user wouldn't be able to troubleshoot easily)